File: packages/Flyingdarts.Persistence/UserProfile.cs
namespace Flyingdarts.Persistence
{
    public class UserProfile : IUserProfile
    {
        public string UserName { get; set; }
        public string Email { get; set; }
        public string Country { get; set; }

        public UserProfile()
        {
            // required for dynamodb
        }

        public UserProfile(string userName, string email, string country)
        {
            UserName = userName;
            Email = email;
            Country = country;
        }

        public static UserProfile Create(string userName, string email, string country)
        {
            return new UserProfile
            {
                UserName = userName,
                Email = email,
                Country = country
            };
        }
    }
}

File: packages/frontend/flutter/api/sdk/flyingdarts_frontend_flutter_api_sdk/lib/src/users/models/user_profile.dart
// ignore_for_file: non_constant_identifier_names

import 'package:equatable/equatable.dart';
import 'package:json_annotation/json_annotation.dart';

part 'user_profile.g.dart';

@JsonSerializable(createToJson: false)
class UserProfile extends Equatable {
  final String UserId;
  final String UserName;
  final String Country;
  final String Email;

  const UserProfile({
    required this.UserId,
    required this.UserName,
    required this.Country,
    required this.Email,
  });

  factory UserProfile.fromJson(Map<String, dynamic> json) => _$UserProfileFromJson(json);

  @override
  List<Object?> get props => [UserId, UserName, Country, Email];
}
